executable-spec/test/Cardano/Ledger/Spec/STS/Chain/Chain/Properties.hs

```diff
@@ -46,3 +47,82 @@ qc_revealsAreClassified
       & dom
       & length
       & fromIntegral
+
+relevantCasesAreCovered :: QC.Property
+relevantCasesAreCovered
+  = QC.withMaxSuccess 300
+  $ STS.Gen.forAllTrace @(CHAIN ShortHash) @() maxTraceLength ()
+  $ \traceSample ->
+      -- 80% of traces should include a [1 - 20] percent of update payload
+      QC.cover 80
+        ( (updatePayloadPct traceSample) >= 1
+        &&
+          (updatePayloadPct traceSample) <= 20
+        )
+        "a reasonable pct of update payload appears in the trace "
+        $
+
+      -- X% of traces should: stake distribution is skewed
+      QC.cover 10
+         (stakeDistIsSkewed traceSample)
+         "stake distribution is skewed"
+         $
+
+      -- X% of traces should: stake distribution is uniform
+      QC.cover 10
+         (stakeDistIsUniform traceSample)
+         "stake distribution is uniform"
+         $
+      -- X% of traces should: Submitted SIPs in the trace are unique
+
+      -- X% of traces should: Submitted SIPs in the trace are not unique
+
+      -- X% of traces should: Submitted SIPs correspond only to valid stake holders
+
+      -- X% of traces should: Submitted SIPs correspond also to invalid stake holders
+
+      -- X% of traces should: for every Submitted SIP there is a Reveal
+
+      -- X% of traces should: there are Submitted SIPs that have not been Revealed yet
+
+      -- X% of traces should: there are Revealed SIPs that have not been submitted
```

Comment:
At the moment the generators forbid this (otherwise the "only valid signals are generated" property won't hold).